app/models/friendship.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class Friendship < ApplicationRecord
+  belongs_to :active_friend, class_name: 'User'
+  belongs_to :passive_friend, class_name: 'User'
+
+  default_scope { order(created_at: :asc) }
+
+  validates :combined_ids, uniqueness: true
+
+  before_validation :concatenate_ids
+  before_validation :prevent_self_invite
```

Comment:
shouldn't it be a custom validator?

Reply:
I haven't made any custom validators in the past. So I opted here for a simple custom method. What is the benefit of creating a custom validator for this?